src/d3a/models/strategy/load_hours.py

```diff
@@ -309,8 +312,13 @@ def event_activate_energy(self):
 
     @property
     def active_markets(self):
-        return [market for market in self.area.all_markets
-                if self._is_market_active(market)]
+        return self._active_markets
+
+    def _calculate_active_markets(self):
+        self._active_markets = [
+            market for market in self.area.all_markets
+            if self._is_market_active(market)
+        ] if self.area else []
```

Comment:
Is this ever going to happen, that self.area is None?

Reply:
This is more of a failsafe, I do not think that the active_markets are accessed during activate event.